packages/core-mobile/app/new/features/portfolio/components/TokenManagementItem.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { FC } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { selectTokenVisibility, toggleTokenVisibility } from 'store/portfolio'
+import { isTokenVisible } from 'store/balance/utils'
+import { LocalTokenWithBalance } from 'store/balance'
+import { Text, Toggle, View } from '@avalabs/k2-alpine'
+import { Space } from 'components/Space'
+import { LogoWithNetwork } from './assets/LogoWithNetwork'
+
+type Props = {
+  token: LocalTokenWithBalance
+}
+
+const TokenManagementItem: FC<Props> = ({ token }) => {
+  const dispatch = useDispatch()
+  const tokenVisibility = useSelector(selectTokenVisibility)
+  const isToggledOn = isTokenVisible(tokenVisibility, token)
+
+  function handleChange(): void {
+    dispatch(
+      toggleTokenVisibility({ tokenId: token.localId, value: !isToggledOn })
```

Comment:
nit: toggleTokenVisibility({ tokenId: token.localId}) since the value can be accessed in redux slice